tensorboard/uploader/uploader_main.py

```diff
@@ -363,8 +374,16 @@ def execute(self, server_info, channel):
                 ("Runs", str(experiment.num_runs)),
                 ("Tags", str(experiment.num_tags)),
             ]
-            for (name, value) in data:
-                print("\t%s %s" % (name.ljust(12), value))
+            if self.json:
+                experiments_json.append(
+                    collections.OrderedDict([("URL", url)] + data)
+                )
+            else:
+                print(url)
+                for (name, value) in data:
+                    print("\t%s %s" % (name.ljust(12), value))
+        if self.json:
+            print(json.dumps(experiments_json, indent=2))
```

Comment:
So the output here looks like this:

```json
[
  {
    "URL": "https://tensorboard.dev/experiment/abcdefghijklmnopqrstuv/",
    "Name": "[No Name]",
    "Description": "[No Description]",
    "Id": "abcdefghijklmnopqrstuv",
    "Created": "2019-10-25 07:22:30",
    "Updated": "2019-10-25 07:22:34",
    "Scalars": "3208",
    "Runs": "8",
    "Tags": "4"
  }
]
```

There are a few problems with this:

 1. The keys are the same as the human-readable form, and in particular
    are capitalized and (after #3464) may have whitespace.
 2. Numeric values like `runs` and `tags` are strings, not ints.
 3. Name and description have dummy values instead of proper `""`s.
 4. The output is slurped and not streamed.

On (1): After #3464, this will have an extra `"Binary object bytes"`
key. But this is pretty unidiomatic. JSON keys do not typically have
whitespace; you can’t use such a key in a `jq` dot-selector:

```
$ printf '{"one": 1, "two three": 23}' | jq '.one'
1
$ printf '{"one": 1, "two three": 23}' | jq '.two three'
jq: error: syntax error, unexpected IDENT, expecting $end (Unix shell quoting issues?) at <top-level>, line 1:
.two three     
jq: 1 compile error
```

On (2): The integers are being converted to string for human
presentation, but should be left as ints in the JSON output for
programmatic manipulation.

On (3): Similar to (2). The dummy values are just to prevent confusing
human-readable output with a blank space. There’s no ambiguity in using
JSON `""`.

On (4): I see that you considered a streaming response but didn’t like
that it was “less easy to parse programmatically”. Is it? You can
convert a streaming response to a buffered response by simply piping
through `jq -s`, but you can’t go the other way around without waiting
for the whole buffer. And as long as we make sure to print one record
per line, people using tools other than JQ should be able to deal with
it handily (`for line in file: json.loads(line)` works in Python, e.g.).

One way to resolve these could be to change `data` from a 2-column table
to a 4-column table with human readable name, JSON key, raw value, and
formatter used when displaying to humans: e.g.,

```
            data = [
                ("Name", "name", experiment.name, lambda x: x or "[No Name]"),
                ("Description", "description", experiment.description, lambda x: x or "[No Description]"),
                ("Id", "id", experiment.experiment_id, str),
                ("Created", "created", util.format_time(experiment.create_time), str),
                ("Updated", "updated", util.format_time(experiment.update_time), str),
                ("Scalars", "scalars", experiment.num_scalars, str),
                ("Runs", "runs", experiment.num_runs, str),
                ("Tags", "tags", experiment.num_tags, str),
            ]
```

and project out the appropriate fields in the JSON and non-JSON cases;
of course feel free to pick a different implementation if you prefer.